crates/nu-std/std/testing.nu

```diff
@@ -1,5 +1,108 @@
 use log.nu
 
+# Here we store the list of annotations the test runner should concern itself with
+# Functions with no annotations or with annotations not on the list are rejected during module evaluation
+# Annotations starting with test- may be used multiple times throughout the module as the function names are stored in a list
+# Other annotations should only be used once within a module file
+# If you find yourself in need of multiple before- or after- functions it's a sign your test suite probably needs redesign
+def valid-annotations [] {
+    [
+        test,
+        test-skip,
+        before-each,
+        before-all,
+        after-each,
+        after-all
```

Comment:
these would translate into
- `# test`
- `# test-skip`
- `# before-each`
- `# before-all`
- `# after-each`
- `# after-all`

but maybe that'd be great to have a clear annotation syntax?
- `#[test]` and `#[test::skip]` a bit like in Rust
- `$[test]` and `$[test skip]` with *command* syntax
- `#[test]` and `#[test]` + `#[ignore]`
- some other mix?

Reply:
The annotations are used as column names during source code parsing so I'd prefer to avoid any characters that might be misinterpreted by nushell

Reply:
> The annotations are used as column names during source code parsing so I'd prefer to avoid any characters that might be misinterpreted by nushell
>
i do not think this should dictate the way the annotations are implemented :open_mouth: 

maybe switching away from the record access would be a good thing, but i think this would work already :smirk: 
e.g. the following works fine
```bash
[
    ["[foo]", "[bar]"];
    [1 2]
    [3 4]
] | get "[foo]"
```
or
```bash
let table = [
    ["[foo]", "[bar]"];
    [1 2]
    [3 4]
]
$table."[foo]"
```

this would simply imply a bit of changes in the source of `std testing` :thinking: 

maybe we can ask people on the Discord to have opinions? :yum:

Reply:
External and internal annotation names are now stored in a map and we can name them whatever we want